cincinnati/src/lib.rs

```diff
@@ -128,6 +128,28 @@ impl Graph {
     pub fn releases_count(&self) -> u64 {
         self.dag.node_count() as u64
     }
+
+    /// Prune the graph from all abstract releases.ConcreteRelease
+    ///
+    /// Return the number of pruned releases
+    pub fn prune_abstract(&mut self) -> usize {
+        let to_remove: Vec<daggy::NodeIndex> = self
+            .dag
+            .node_references()
+            .filter_map(|nr| {
+                if let Release::Abstract(_) = nr.weight() {
+                    Some(nr.0)
+                } else {
+                    None
+                }
+            })
+            .collect();
```

Comment:
Out of curiosity, what's the reason for collecting here and doing a further loop instead of filtering nodes during iteration?

Reply:
I couldn't find a way to remove the nodes while iterating over them. Removing them requires exclusive access to `self.dag` but it's borrowed by the iteration.

Reply:
Ack, let's leave it as is for now then.